modules/social_features/social_out_of_office/social_out_of_office.module

```diff
@@ -0,0 +1,256 @@
+    $current_user = \Drupal::currentUser()->getAccount();
+    $user = $variables['elements']['#user'];
+
+    if ($current_user->id() !== $user->id()) {
+      /** @var \Drupal\profile\ProfileStorageInterface $profile_storage */
+      $profile_storage = \Drupal::entityTypeManager()->getStorage('profile');
+
+      /** @var \Drupal\profile\Entity\Profile $profile */
+      $profile = $profile_storage->loadByUser($user, SocialOutOfOfficeHelperInterface::PROFILE_TYPE);
+
+      if ($profile instanceof ProfileInterface) {
+        /** @var \Drupal\social_out_of_office\SocialOutOfOfficeHelperInterface $ooo_helper */
+        $ooo_helper = \Drupal::service('social_out_of_office.helper');
+
+        // Check if user is out of office.
+        if ($ooo_helper->isUserOutOfOffice($profile)) {
+          // Retrieve out of office dates.
+          $dates = $ooo_helper->getOutOfOfficeDates($profile);
+
+          $suffix = ' ' . t('is out of office from @start_date to @end_date', [
+            '@start_date' => $dates['start'],
+            '@end_date' => $dates['end'],
+          ]);
+          $suffix .= $variables['content']['linked_username']['#suffix'];
+          $variables['content']['linked_username']['#suffix'] = $suffix;
+        }
+      }
+
+      // Add OoO cache tag.
+      $variables['#cache']['tags'][] = 'ooo_dates:uid:' . $user->id();
+    }
+  }
+}
+
+/**
+ * Implements hook_entity_presave().
+ */
+function social_out_of_office_entity_presave(EntityInterface $entity): void {
+  // Show message about users that out of office if someone mentioned them in
+  // post or comment.
+  if (
+    !($entity instanceof PostInterface) &&
+    !($entity instanceof CommentInterface)
+  ) {
+    return;
+  }
+
+  $post_comment_related_users = [];
+
+  // Comment entity type.
+  if ($entity instanceof CommentInterface) {
+    $body_value = $entity->get('field_comment_body')->getValue();
+
+    // If commented entity author is OoO we should show message about that.
+    /** @var \Drupal\social_post\Entity\PostInterface $commented_entity */
+    /** @var \Drupal\node\NodeInterface $commented_entity */
+    $commented_entity = $entity->getCommentedEntity();
+    if ($commented_entity !== NULL) {
+      $post_comment_related_users[] = $commented_entity->getOwnerId();
+    }
+
+    // If parent comment author is OoO we should show message about that.
+    $parent_comment = $entity->getParentComment();
+    if ($parent_comment) {
+      $post_comment_related_users[] = $parent_comment->getOwnerId();
+    }
+  }
+  // Post entity type.
+  else {
+    $body_value = $entity->get('field_post')->getValue();
+  }
+
+  /** @var \Drupal\social_out_of_office\SocialOutOfOfficeHelperInterface $ooo_helper */
+  $ooo_helper = \Drupal::service('social_out_of_office.helper');
+
+  /** @var \Drupal\social_mentions\SocialMentionsHelperInterface $social_mentions_helper */
+  $social_mentions_helper = \Drupal::service('social_mentions.helper');
```

Comment:
This will requires some if mentions is enabled/available

Reply:
I've added defensive check.